### ISSUE TITLE
Weight_decay in torch.Adam

### ISSUE THREAD

**Issue by maridia** [pytorch/pytorch#48793]
## 📚 Documentation

In the current [pytorch docs for torch.Adam](https://pytorch.org/docs/stable/optim.html), the following is written:

"Implements Adam algorithm.

It has been proposed in [Adam: A Method for Stochastic Optimization](https://arxiv.org/abs/1412.6980). The implementation of the L2 penalty follows changes proposed in [Decoupled Weight Decay Regularization](https://arxiv.org/abs/1711.05101)."

This would lead me to believe that the current implementation of Adam is essentially equivalent to AdamW.  The fact that torch.AdamW exists as a separate optimizer leads me to believe that this isn't true.  Also, after looking at the source code of torch.Adam, I don't see any difference from a standard L2 penalty implementation.  So is the documentation incorrect and the "changes proposed in [Decoupled Weight Decay Regularization]" are actually absent from torch.Adam?

cc @jlin27 @mruberry @vincentqb

**Comment by ptrblck** [comment]
Additional context: issue was discussed [here](https://discuss.pytorch.org/t/weight-decay-in-torch-adam/104942/4?u=ptrblck) and the author would be interested in creating a PR with some support.
If the change is applicable, feel free to ping me to discuss how to create the PR.

**Comment by agamemnonc** [comment]
Could someone please confirm whether this is indeed a documentation issue (i.e.,  `Adam` uses the default weight_decay rule) or, conversely, if the two implementations of `Adam` and `AdamW` are equivalent?

**Comment by jan992** [comment]
To me it looks like a documentation issue. The weight decay in `Adam` is [implemented as](https://github.com/pytorch/pytorch/blob/a389b30bfc12a22f12686c7be09693abd9604d80/torch/optim/functional.py#L83)

```python
grad = grad.add(param, alpha=weight_decay)
```
which to me appears to be normal L2-regularization. Or am I getting something wrong here?

**Comment by apillay-cmu** [comment]
I see that the issue has been fixed with [#50464](https://github.com/pytorch/pytorch/pull/50464). Can this issue be closed in that case?

**Comment by janeyx99** [comment]
yep! closing as the docs should be correct now
